src/MSBuild/XMake.cs

```diff
@@ -3381,13 +3493,21 @@ private static string ProcessToolsVersionSwitch(string[] parameters)
             var outVerbosity = verbosity;
             ProcessBinaryLogger(binaryLoggerParameters, loggers, ref outVerbosity);
 
-            ProcessLoggerSwitch(loggerSwitchParameters, loggers, verbosity);
+            // When returning the result of evaluation from the command line, do not use custom loggers.
+            if (!useSimpleErrorLogger)
+            {
+                ProcessLoggerSwitch(loggerSwitchParameters, loggers, verbosity);
+            }
```

Comment:
This means that if one of the new options is specified, logger switches are silently ignored. I'd say that's a bug and should be fixed in this PR or tracked in an issue.

Reply:
To be clear, this is silently ignoring _custom_ logger switches, not all logger switches. That was an intentional choice. As I recall, there was concern about polluting output streams with different kinds of data—that is, if you have a custom logger that writes information to stdout, and I try to put the json in stdout, they could get mixed up (or at least be next to each other), which would make parsing it confusing.

Reply:
I understand it was intentional for some of the existing switches to be mutually exclusive with the new ones. It's surprising that we would intentionally want to ignore them instead of throwing an error, though. Can you please confirm this is really the case?

Reply:
That was my understanding, but there have been a couple cases of misunderstandings here, so @baronfel?

Reply:
Correct - we opted to silently ignore for a few reasons:

* The custom loggers are entirely ignored when the new flags are used to prevent stdout/stderr hijacking
* MSBuild is still performing a build, and so reading arguments from not-only-commandline sources like Directory.Build.rsp can mean that tooling/scripts may not be in complete control of the final command line. So we need to be able to use this functionality even on the presence of conflicting command lines, so we just ignore instead of error.

Reply:
Thank you for following up. I stand corrected then. It would be super awesome for this logic to be covered with tests or at least commented so the future me wouldn't try to "fix it". (Extra points!)